File: prog-7-1.cpp
#include <iostream>
using namespace std;

int digit_add(long long num, int dig)
{
	while (num >= dig)
	{
		dig *= 10;
	}

	return dig + num;
}

bool func (long long N, long long sum, long long head, long long tail)
{
	if (head == 0)
	{
		if (sum + tail == N)
			cout << tail;

		return sum + tail == N;
	}

	if (func (N, sum + tail, head / 10, head % 10))
	{
		cout << "+" << tail;
	}
	else if (func (N, sum - tail, head / 10, head % 10))
	{
		cout << "-" << tail;
	}
	else if (func (N, sum, head / 10, digit_add(tail, head % 10)))
	{}
	else
	{
		return false;
	}

	return true;
}

int main()
{
	long long N, digits;
	cin >> N >> digits;
	func (N, 0, digits /10, digits % 10);
}

File: prog-7-1.hs
digitAdd :: Integer -> Integer -> Integer
digitAdd num dig
    | num >= dig = digitAdd num (dig * 10)
    | otherwise = dig + num

func :: Integer -> Integer -> Integer -> Integer -> Bool
func n sum head tail
    | head == 0 = if sum + tail == n then True else False
    | func n (sum + tail) (head `div` 10) (head `mod` 10) = True
    | func n (sum - tail) (head `div` 10) (head `mod` 10) = True
    | func n sum (head `div` 10) (digitAdd tail (head `mod` 10)) = True
    | otherwise = False

main :: IO ()
main = do
    input <- getLine
    let [n, digits] = map read $ words input
    func n 0 (digits `div` 10) (digits `mod` 10)
 
